docs/site/Include-filter.md

```diff
@@ -0,0 +1,240 @@
+    address: {'8200 Warden Ave'}
+ },
+ {...}
+]
+```
+
+- Return all customers with their orders and also the shipment information of orders:
+
+{% include code-caption.html content="Node.js API" %}
+
+```ts
+await customerRepository.find({
+  include: [
+    {
+      relation: 'orders',
+      scope: {
+        include: [{relation: 'shipment'}],
+      },
+    },
+  ],
+});
+```
+
+{% include code-caption.html content="REST" %} 
+
+(using `encodeURIComponent`)
+
+```
+GET /customers?filter=%7B"include"%3A%5B%7B"relation"%3A"orders"%2C"scope"%3A%7B"include"%3A%5B%7B"relation"%3A"shipment"%7D%5D%7D%7D%5D%7D
+```
+
+Result:
+
+```ts
+[{
+    id: 1,
+    name: 'Tom Nook',
+    orders:[
+        {id: 123,
+        desc: 'pencil',
+        shipment: {id: 999, company: 'UPS'} // nested related models
+        }
+    ],
+ },
+ {...}
+]
+```
+
+#### Combined use of `fields` and `include` for a `belongsTo` relation
+
+If you want to use both `include` and [`fields`](Fields-filter.md) to display only specific fields of a model and a specific belongsTo relation, **you need to add the relation foreign key in the `fields`** :
+
+Return all posts only with field `title` and the relation `category`:
+```ts
+await postRepository.find({include: [{relation: 'category'}], fields: ['title', 'categoryId'});
+```
+
+#### Include with filters
+
+In some cases, you may want to apply filters to related models to be included.
+
+{% include note.html content="
+
+When you apply filters to related models, the query returns results from the first model plus any results from related models with the filter query,
+similar to a \"left join\" in SQL.
+
+" %}
+
+LoopBack supports that with the following syntax (for example):
+
+```ts
+await postRepository.find({
+  include: [{
+    relation: 'owner', // include the owner object
+    scope: { // further filter the owner object
+      fields: ['username', 'email'], // only show two fields
+      include: { // include orders for the owner
+        relation: 'orders', 
+        scope: {
+          where: {orderId: 5} // only select order with id 5
+        }
+      }
+    }
+  }]
+  });
+```
+
+For real-world scenarios where only users in `$authenticated` or `$owner` roles should have access, use `findById()`.
+For example, the following example uses filters to perform pagination:
+
+```ts
+await postRepository.findById('123', {
+  include: [{
+    relation: 'owner',
+    scope: { // fetch 1st "page" with 5 entries in it
+      skip:0,
+      limit:5
+    }
+  }]
+});
+```
+
+#### Access included objects
+
+In the Node.js API, call `toJSON()` to convert the returned model instance with related items into a plain JSON object. For example:
+
+```ts
+const result = await postRepository.find({include: [{relation: 'owner'}, {relation: 'orders'}]});
+console.log(result.owner.posts, result.owner.orders);
+ //... 
+```
+
+Note the relation properties such as `post.owner` reference a JavaScript **function** for the relation method.
```

Comment:
Please explain what happens when you call those functions.

Reply:
Reworded!